crates/provider-sdk/src/provider.rs

```diff
@@ -182,147 +224,173 @@ impl ProviderConnection {
         self.rpc_client.flush().await
     }
 
-    /// Returns the nats rpc topic for capability providers
-    pub fn provider_rpc_topic(&self) -> String {
-        format!(
-            "wasmbus.rpc.{}.{}.{}",
-            &self.lattice, &self.host_data.provider_key, self.host_data.link_name
-        )
-    }
-
     /// Subscribe to a nats topic for rpc messages.
     /// This method starts a separate async task and returns immediately.
     /// It will exit if the nats client disconnects, or if a signal is received on the quit channel.
     pub async fn subscribe_rpc<P>(
         &self,
         provider: P,
-        mut quit: QuitSignal,
+        quit: QuitSignal,
         lattice: String,
-    ) -> ProviderInitResult<JoinHandle<()>>
+        provider_key: impl AsRef<str>,
+    ) -> ProviderInitResult<Vec<JoinHandle<Result<()>>>>
     where
         P: Provider + Clone,
     {
-        let mut sub = self
-            .rpc_client
-            .client()
-            .queue_subscribe(
-                self.provider_rpc_topic(),
-                RPC_SUBSCRIPTION_QUEUE_GROUP.to_string(),
-            )
-            .await?;
-        let this = self.clone();
-        let handle = tokio::spawn(async move {
-            loop {
+        let mut handles = Vec::new();
+        let provider_key = provider_key.as_ref();
+
+        // Build a wrpc client that we can use to listen for incoming invocations
+        let wrpc_client = WrpcNatsClient::new(
+            self.rpc_client.client(),
+            format!("{lattice}.{}", provider_key),
+        );
+        let link_name = self.host_data.link_name.clone();
```

Comment:
nit: We should be able to get rid of all mentions to link name, since we're always passing "default" now from the host

Reply:
ah but that's not always going to be the case though, right -- this code shouldn't break when we use actual link names, either.

Reply:
Ah yeah I mean that we can remove link name from the host data, but you're right that we need to keep them around when we use interface link definitions with link names